### PR TITLE
Add factory reset cmd

### PR DESCRIPTION
**Pull request by Junnplus**
Signed-off-by: Ye Sijun <junnplus@gmail.com>

Add `limactl factory-reset` command for wiping the data of instance.

### CODE REVIEW

**Review by AkihiroSuda — DISMISSED** [review]
Thanks

**Comment by jandubois** [comment]
Thanks!

I'm wondering if we should have a way to factory-reset an instance without getting prompted to restart it (like `--tty=false` for the `start` command).

This is a bit ugly:

```console
$ l factory-reset </dev/null
INFO[0000] The QEMU process seems already stopped
INFO[0000] The host agent process seems already stopped
INFO[0000] Removing *.pid *.sock under "/Users/jan/.lima/default"
INFO[0000] Instance "default" has been factory reset
? Do you want to start the instance now?  (Y/n) FATA[0000] EOF
^[[60;156R^[[60;49R
```

**Comment by Junnplus** [comment]
> I'm wondering if we should have a way to factory-reset an instance without getting prompted to restart it (like --tty=false for the start command).

Unlike start command, we cannot determine if the user wants to start the instance after a factory reset.
The simple solution is to remove the ask, use the `--start=true` flag.

File: cmd/limactl/factory-reset.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	for _, f := range fi {
+		path := filepath.Join(inst.Dir, f.Name())
+		if !strings.HasSuffix(path, ".yaml") && !strings.HasSuffix(path, ".yml") {
+			logrus.Infof("Removing %q", path)
+			if err := os.Remove(path); err != nil {
+				logrus.Error(err)
+			}
+		}
+	}
+	logrus.Infof("Instance %q has been factory reset", instName)
```
Comment (AkihiroSuda) [inline cmd/limactl/factory-reset.go:57]: We can just exit here. No need to start, as factory-reset would be very rarely used command.

Comment (Junnplus) [reply cmd/limactl/factory-reset.go:55]: SGTM.

**Review by AkihiroSuda — APPROVED** [review]

**Review by jandubois — APPROVED** [review]
Thanks, LGTM

**Comment by AkihiroSuda** [comment]
Next time please squash commits 🙏

**Pull request merged**